rainier-core/src/main/scala/com/stripe/rainier/sampler/MAP.scala

```diff
@@ -1,27 +1,27 @@
 package com.stripe.rainier.sampler
 
-import com.stripe.rainier.compute._
-
 final case class MAP(stepSize: Double) extends Sampler {
-  def sample(context: Context,
+  def sample(densityFunction: DensityFunction,
              warmupIterations: Int,
              iterations: Int,
              keepEvery: Int)(implicit rng: RNG): List[Array[Double]] = {
-    val values = MAP.optimize(context, warmupIterations, stepSize)
+    val values = MAP.optimize(densityFunction, warmupIterations, stepSize)
     List(values)
   }
 }
 
 object MAP {
-  def optimize(context: Context,
+  def optimize(densityFunction: DensityFunction,
                iterations: Int,
                stepSize: Double): Array[Double] = {
-    val cf =
-      context.compiler
-        .compile(context.variables, context.density :: context.gradient)
-    val initialValues = context.variables.map { v =>
-      0.0
-    }.toArray
+    val cf = (params: Array[Double]) =>
+      densityFunction.density(params) +: densityFunction.gradient(params)
```

Comment:
you're building this list here and then taking it apart again on line 27. could just simplify by calling densityFunction.gradient directly there.

Reply:
👍